### PR TITLE
update(build): update libcurl to 7.87.0

### PR DESCRIPTION
**Pull request by LucaGuerra**
Signed-off-by: Luca Guerra <luca@guerra.sh>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

**Does this PR require a change in the driver versions?**

No

**What this PR does / why we need it**:

Update libcURL to the latest version, fixing 7 potential security vulnerabilities (none of them critically impacting Falco as far as I understand) disclosed in the last months, including CVE-2022-43551 and CVE-2022-43552 disclosed today. See: https://curl.se/docs/security.html .

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by Andreagit97 — APPROVED** [review]
/approve

File: userspace/libsinsp/mesos_http.cpp [hunk 1]
```diff
@@ -539,7 +539,7 @@ int mesos_http::get_socket(long timeout_ms)
 
 		check_error(curl_easy_perform(m_select_curl));
 
-		check_error(curl_easy_getinfo(m_select_curl, CURLINFO_LASTSOCKET, &sockextr));
+		check_error(curl_easy_getinfo(m_select_curl, CURLINFO_ACTIVESOCKET, &sockextr));
```
Comment (FedeDP) [inline userspace/libsinsp/mesos_http.cpp:542]: Mind to explain this change? :)

Comment (LucaGuerra) [reply userspace/libsinsp/mesos_http.cpp:542]: Of course, look at the description: https://curl.se/libcurl/c/CURLINFO_LASTSOCKET.html . In fact, that is now going to emit a warning!

**Review by FedeDP — APPROVED** [review]
/approve
Thanks!

**Pull request merged**